### PR TITLE
Remove posix calls in TemporaryFile.swift

### PR DESCRIPTION
**Pull request by a20012251**
This implements a portable version of mkstemps and mkdtemp so that the code also works on Windows. This simplified the code considerably

### CODE REVIEW

**Comment by jmittert** [comment]
This looks good to me, and the removeItem that this uses just got implemented on Windows today (https://github.com/apple/swift-corelibs-foundation/pull/2117).

@aciidb0mb3r

File: Sources/Basic/TemporaryFile.swift [hunk 1]
```diff
@@ -45,18 +29,52 @@ private extension TempFileError {
...
+let ALPHANUMERIC_CHARS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+func createMkstempLikeName(prefix: String, suffix: String) -> String {
+    let randomMiddleStringLen = 6
+    let randomMiddleString =
+        String((0..<randomMiddleStringLen).map { _ in ALPHANUMERIC_CHARS.randomElement()! })
+    return prefix + "." + randomMiddleString + suffix
+}
+
+func mkstemps(dir: AbsolutePath, prefix: String, suffix: String) throws -> AbsolutePath {
```
Comment (aciidb0mb3r) [inline Sources/Basic/TemporaryFile.swift]: I am a little nervous about this. I think we should at least use mkstemps where it's available.

Comment (a20012251) [reply Sources/Basic/TemporaryFile.swift]: I think that the logic is quite straightforward.
Why are you nervous about it? We can discuss it

Comment (a20012251) [reply Sources/Basic/TemporaryFile.swift]: My team changed my mind, i'll implement this in a posix way and additionally a windows specific way

**Comment by tomerd** [comment]
@compnerd is this still relevant given your recent work for Windows support?

**Comment by compnerd** [comment]
I don't think that this is relevant any longer.  @a20012251 if you think that this is still relevant, please rebase and reopen the PR.

**Pull request closed without merging**